origin-discovery/src/apollo/resolvers.js

```diff
@@ -93,21 +106,32 @@ const resolvers = {
     price (offer) {
       return { currency: 'ETH', amount: offer.priceEth }
     },
-    listing (offer, args, context, info) {
+    async listing (offer, args, context, info) {
       const requestedSubFields = info.fieldNodes[0].selectionSet.selections
-      const isIdOnly = requestedSubFields.filter(x => x.name.value !== 'id').length === 0
-      if (isIdOnly) {
+      const idsOnly = requestedSubFields.filter(x => x.name.value !== 'id').length === 0
+      if (idsOnly) {
         return { id: offer.listingId }
       } else {
-        return search.Listing.get(offer.listingId)
+        const row = await db.Listing.findByPk(offer.listingId)
+        return row.data
```

Comment:
We could handle the case when a wrong/not existing `listingId` is queried and return null instead of throwing an exception.

Reply:
We could though if the listing is not found, it does indicate there is a problem... it should not be possible that an offer exists but there is no associated listing in the DB.
I guess we should discuss about our overall GraphQL error handling. If the back-end errors out for a valid reason (ex: data should be there but is not), do we want to mask that by returning no data or do we want to surface the exception ? (I'm leaning towards the latter, to help us investigating issues).

Reply:
Ah yes, good point it is tied to an offer. Yes I agree, an exception is a good way to respond here. Perhaps a custom exception message? Something like `Listing of the offer with id: ${offerId} could not be found!` instead of "Can not read property `data` of undefined"